pylops/LinearOperator.py

```diff
@@ -47,32 +48,211 @@ class LinearOperator(spLinearOperator):
     Parameters
     ----------
     Op : :obj:`scipy.sparse.linalg.LinearOperator` or :obj:`scipy.sparse.linalg._ProductLinearOperator` or :obj:`scipy.sparse.linalg._SumLinearOperator`
-        Operator
+        Operator. If other arguments are provided, they will overwrite those obtained from ``Op``.
+    shape : :obj:`tuple(int, int)`, optional
+        Shape of operator. If not provided, obtained from ``dims`` and ``dimsd``.
+    dims : :obj:`tuple(int, ..., int)`, optional
+        .. versionadded:: 2.0.0
+
+        Dimensions of model. If not provided, ``(self.shape[1],)`` is used.
+    dimsd : :obj:`tuple(int, ..., int)`, optional
+        .. versionadded:: 2.0.0
+
+        Dimensions of data. If not provided, ``(self.shape[0],)`` is used.
     explicit : :obj:`bool`, optional
         Operator contains a matrix that can be solved explicitly
-        (``True``) or not (``False``)
+        (``True``) or not (``False``). Defaults to ``False``.
     clinear : :obj:`bool`, optional
         .. versionadded:: 1.17.0
 
-        Operator is complex-linear.
+        Operator is complex-linear. Defaults to ``True``.
     name : :obj:`str`, optional
         .. versionadded:: 2.0.0
 
         Name of operator (to be used by :func:`pylops.utils.describe.describe`)
 
     """
 
-    def __init__(self, Op=None, explicit=False, clinear=None, name=None):
-        self.explicit = explicit
+    def __init__(
+        self,
+        Op=None,
+        dtype=None,
+        shape=None,
+        dims=None,
+        dimsd=None,
+        clinear=None,
+        explicit=None,
+        name=None,
+    ):
         if Op is not None:
             self.Op = Op
-            self.shape = self.Op.shape
-            self.dtype = self.Op.dtype
-            self.clinear = getattr(self.Op, "clinear", True)
+            # All Operators must have shape and dtype
+            dtype = self.Op.dtype if dtype is None else dtype
+            shape = self.Op.shape if shape is None else shape
+            # Optionally, some operators have other attributes
+            dims = getattr(Op, "dims", (Op.shape[1],)) if dims is None else dims
+            dimsd = getattr(Op, "dimsd", (Op.shape[0],)) if dimsd is None else dimsd
+            clinear = getattr(Op, "clinear", True) if clinear is None else clinear
+            explicit = (
+                getattr(self.Op, "explicit", False) if explicit is None else explicit
+            )
+            if explicit and hasattr(Op, "A"):
+                self.A = Op.A
+            name = getattr(Op, "name", None) if name is None else name
+
+        if dtype is not None:
+            self.dtype = dtype
+        if shape is not None:
+            self.shape = shape
```

Comment:
since it is a property now should it not be `self._shape`? Same for all others...

Reply:
I thought about this, but I opted to keep using the setter, since has checks/coercions to ensure the values are "legal". For example, setting `self._shape = [3, 2]` would be acceptable, but `self.shape = [3, 2]` is nicer since it would coerce into a tuple (`self._shape == (3, 2)`).

Reply:
Right, I see.